torch/csrc/jit/script/module.h

```diff
@@ -125,17 +126,117 @@ struct TORCH_API Method {
   }
 
  private:
-  // Methods are uniqued onwed by a single module. This raw pointer allows
-  // looking up the module.
-  ModulePtr owner_;
+  // Methods are uniqued onwed by a single object. This raw pointer allows
+  // looking up the object.
+  ObjectPtr owner_;
 
   // Underlying unbound function
   // This is the _lowered_ function and is different than the
   // first-class function in class_compilation_unit()
   Function* function_;
 };
 
-struct TORCH_API Module {
+struct TORCH_API Object {
+  Object() {}
+  Object(ObjectPtr object_value) : object_value_(std::move(object_value)) {}
+
+  const c10::QualifiedName& name() const {
```

Comment:
We should kill this method and replace it with `obj.type()->name()` in its uses. Asking for the name of an object doesn't really make sense.